scripts/windows/wix-populate_internal_artifacts.py

```diff
@@ -0,0 +1,59 @@
+"""Add all components (files) and directories generated by pyinstaller in
+dist/papis/_internal into the XML manifest that is used to create the MSI."""
+
+import os
+import lxml.etree as et
+import uuid
+
+
+def get_id(*path_components: str) -> str:
+    """Create an element ID based on the WiX constraints:"""
+    path = os.path.join(*path_components)
+    result = f"_{str(abs(hash(path)))}"
+    return result
+
+
+wxs_tree = et.parse(".\\scripts\\windows\\papis_template.wxs")
+wxs_root = wxs_tree.getroot()
+
+xpath_ns = {"main": "http://wixtoolset.org/schemas/v4/wxs"}
+component = wxs_root.xpath(
+    ".//main:Directory[@Id='INSTALLFOLDER']", namespaces=xpath_ns
+)[0]
+package = wxs_root.xpath(".//main:Package", namespaces=xpath_ns)[0]
+
+root_dir = "dist\\papis\\_internal"
+
+directories = et.SubElement(
+    component, "Directory", Name="_internal", Id=get_id(root_dir)
+)
+component_group = et.SubElement(package, "ComponentGroup", Id="_internal")
+
+for root, dirs, files in os.walk(root_dir):
+    for dir in dirs:
+        relpath = os.path.relpath(root, root_dir)
+        parent_path = os.path.dirname(relpath)
+        if parent_path == "":
+            parent_element = directories
+        else:
+            parent_element_id = get_id(root_dir, parent_path)
+            xpath = f".//Directory[@Id='{parent_element_id}']"
+            parent_element = directories.xpath(xpath)[0]
+
+        et.SubElement(parent_element, "Directory", Name=dir, Id=get_id(root, dir))
+
+    if files:
+        for index, file in enumerate(files):
+            component = et.SubElement(
+                component_group,
+                "Component",
+                Guid=str(uuid.uuid4()),
+                Id=f"c{get_id(root, file)}",
+                Directory=get_id(root),
+            )
+            et.SubElement(
+                component, "File", KeyPath="yes", Source=os.path.join(root, file)
+            )
+
+with open("scripts\\windows\\papis.wxs", mode="wb") as f:
```

Comment:
```suggestion
with open("scripts\\windows\\papis.wxs", mode="w") as f:
```
? Why the `b` in there?

Reply:
The XML might look like a text file, but i would get an error saying that the file writer expected writing bytes, not a string. So I complied.